.werft/jobs/build/deploy-to-preview-environment.ts

```diff
@@ -659,22 +661,12 @@ async function addDNSRecord(werft: Werft, namespace: string, domain: string, isL
         }
         werft.log(installerSlices.DNS_ADD_RECORD, "Get ws-proxy loadbalancer IP: " + wsProxyLBIP);
     } else {
-        wsProxyLBIP = getCoreDevIngressIP()
+        wsProxyLBIP = coreDevIngressIP
     }
 
-    var cmd = `set -x \
-    && cd /workspace/.werft/dns \
-    && rm -rf .terraform* \
-    && export GOOGLE_APPLICATION_CREDENTIALS="${GCLOUD_SERVICE_ACCOUNT_PATH}" \
-    && terraform init -backend-config='prefix=${namespace}' -migrate-state -upgrade \
-    && terraform apply -auto-approve \
-        -var 'dns_zone_domain=gitpod-dev.com' \
-        -var 'domain=${domain}' \
-        -var 'ingress_ip=${getCoreDevIngressIP()}' \
-        -var 'ws_proxy_ip=${wsProxyLBIP}'`;
-
-    werft.log(installerSlices.DNS_ADD_RECORD, "Terraform command for create dns record: " + cmd)
-    exec(cmd, { ...metaEnv(), slice: installerSlices.DNS_ADD_RECORD });
+    createDNSRecord(domain, 'gitpod-dev-com', coreDevIngressIP, installerSlices.DNS_ADD_RECORD)
```

Comment:
If you want to run all of these in parallel and then wait for them all to finish (or fail if one of them does) you can use [Promise.all ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)

```ts
await Promise.all([
  createDNSRecord(domain, 'gitpod-dev-com', coreDevIngressIP, installerSlices.DNS_ADD_RECORD),
  createDNSRecord(`*.${domain}`, 'gitpod-dev-com', coreDevIngressIP, installerSlices.DNS_ADD_RECORD),
  createDNSRecord(`*.ws-dev.${domain}`, 'gitpod-dev-com', wsProxyLBIP, installerSlices.DNS_ADD_RECORD),
])
```